pkg/config/validation/validation.go

```diff
@@ -3308,3 +3325,126 @@ func validateNetwork(network *meshconfig.Network) (errs error) {
 func (aae *AnalysisAwareError) Error() string {
 	return aae.Msg
 }
+
+// ValidateTelemetry validates a Telemetry.
+var ValidateTelemetry = registerValidateFunc("ValidateTelemetry",
+	func(cfg config.Config) (Warning, error) {
+		spec, ok := cfg.Spec.(*telemetry.Telemetry)
+		if !ok {
+			return nil, fmt.Errorf("cannot cast to service entry")
+		}
+
+		errs := Validation{}
+
+		errs = appendValidation(errs,
+			validateWorkloadSelector(spec.Selector),
+			validateTelemetryMetrics(spec.Metrics),
+			validateTelemetryTracing(spec.Tracing),
+			validateTelemetryAccessLogging(spec.AccessLogging),
+		)
+		return errs.Unwrap()
+	})
+
+func validateTelemetryAccessLogging(logging []*telemetry.AccessLogging) (v Validation) {
+	if len(logging) > 1 {
+		v = appendWarningf(v, "multiple accessLogging is not currently supported")
+	}
+	for idx, l := range logging {
+		if l == nil {
+			continue
+		}
+		if len(l.Providers) > 1 {
+			v = appendValidation(v, Warningf("accessLogging[%d]: multiple providers is not currently supported", idx))
+		}
+		v = appendValidation(v, validateTelemetryProviders(l.Providers))
+	}
+	return
+}
+
+func validateTelemetryTracing(tracing []*telemetry.Tracing) (v Validation) {
+	if len(tracing) > 1 {
+		v = appendWarningf(v, "multiple tracing is not currently supported")
+	}
+	for _, l := range tracing {
+		if l == nil {
+			continue
+		}
+		if len(l.Providers) > 1 {
+			v = appendWarningf(v, "multiple providers is not currently supported")
+		}
+		v = appendValidation(v, validateTelemetryProviders(l.Providers))
+		if l.RandomSamplingPercentage.GetValue() < 0 || l.RandomSamplingPercentage.GetValue() > 100 {
+			v = appendErrorf(v, "randomSamplingPercentage must be in range [0.0, 100.0]")
+		}
+		for name, tag := range l.CustomTags {
+			if name == "" {
+				v = appendErrorf(v, "tag name may not be empty")
+			}
+			switch t := tag.Type.(type) {
+			case *telemetry.Tracing_CustomTag_Literal:
+				if t.Literal.GetValue() == "" {
+					v = appendErrorf(v, "literal tag value may not be empty")
+				}
+			case *telemetry.Tracing_CustomTag_Header:
+				if t.Header.GetName() == "" {
+					v = appendErrorf(v, "header tag name may not be empty")
+				}
+			case *telemetry.Tracing_CustomTag_Environment:
+				if t.Environment.GetName() == "" {
+					v = appendErrorf(v, "environment tag name may not be empty")
+				}
+			}
+		}
+	}
+	return
+}
+
+func validateTelemetryMetrics(metrics []*telemetry.Metrics) (v Validation) {
+	if len(metrics) > 1 {
+		v = appendWarningf(v, "multiple metrics is not currently supported")
```

Comment:
do we want this restriction to start?  unlike tracing, for metrics we can have multiple providers at the same time (prom and stackdriver).

Reply:
I intentionally made this a warning, not an error. So it won't block; same with tracing and logging.

I guess for metrics it currently supports *nothing* (not yet implemented though), and plausibly the initial implementation will support multiple providers at the same time, so maybe this one doesn't make sense. if anything, we could add a warning that Telemetry is not even implemented yet

Reply:
fair enough. we'll want to add addressing this warning in the alpha promotion plan.